parsl/dataflow/strategy.py

```diff
@@ -275,3 +277,132 @@ def _strategy_simple(self, status_list, tasks, *args, kind=None, **kwargs):
             else:
                 # logger.debug("Strategy: Case 3")
                 pass
+
+    def _strategy_htex(self, tasks, *args, kind=None, **kwargs):
+        """Peek at the DFK and the executors specified.
+
+        We assume here that tasks are not held in a runnable
+        state, and that all tasks from an app would be sent to
+        a single specific executor, i.e tasks cannot be specified
+        to go to one of more executors.
+
+        Args:
+            - tasks (task_ids): Not used here.
+
+        KWargs:
+            - kind (Not used)
+        """
+
+        for label, executor in self.dfk.executors.items():
+            if not executor.scaling_enabled:
+                continue
+
+            # Tasks that are either pending completion
+            active_tasks = executor.outstanding
+
+            status = executor.status()
+            self.unset_logging()
+
+            # FIXME we need to handle case where provider does not define these
+            # FIXME probably more of this logic should be moved to the provider
+            min_blocks = executor.provider.min_blocks
+            max_blocks = executor.provider.max_blocks
+            if isinstance(executor, HighThroughputExecutor):
+
+                tasks_per_node = executor.workers_per_node
+            elif isinstance(executor, ExtremeScaleExecutor):
+                tasks_per_node = executor.ranks_per_node
+
```

Comment:
add default behaviour (error?) when neither of the above cases fire and task_per_node is left undefined